src/rws_client.cpp

```diff
@@ -147,9 +147,20 @@ RWSClient::RWSResult RWSClient::getMechanicalUnitJointTarget(const std::string&
   return evaluatePOCOResult(httpGet(uri), evaluation_conditions);
 }
 
-RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string& mechunit)
+RWSClient::RWSResult RWSClient::getMechanicalUnitRobTarget(const std::string &mechunit,
+                                                           const std::string &tool,
+                                                           const std::string &wobj,
+                                                           const std::string &coordinate)
 {
   std::string uri = generateMechanicalUnitPath(mechunit) + Resources::ROBTARGET;
+  if (!coordinate.empty())
+  {
+    uri += "?coordinate=" + coordinate;
+    if (!tool.empty())
+      uri += "&tool=" + tool;
+    if (!wobj.empty())
+      uri += "&wobj=" + wobj;
+  }
```

Comment:
According to the [RWS documentation](http://developercenter.robotstudio.com/blobproxy/devcenter/Robot_Web_Services/html/ms_mechunit_robtarget_page.html), then the only valid values of the `coordinate` argument are `Base`, `Word`, `Tool` or `Wobj`.

I would suggest to, for example, add an `Coordinates` enum (to [rws_client.h](https://github.com/ros-industrial/abb_librws/blob/4741d83b865a1a495d9973294607d65c660d2f21/include/abb_librws/rws_client.h)), and replace the string data type of the `coordinate` argument with the enum. And then also add something like this (to [rws_common.h](https://github.com/ros-industrial/abb_librws/blob/4741d83b865a1a495d9973294607d65c660d2f21/include/abb_librws/rws_common.h) and [rws_common.cpp](https://github.com/ros-industrial/abb_librws/blob/4741d83b865a1a495d9973294607d65c660d2f21/src/rws_common.cpp)), which can be used when mapping the enum to a string:
```
const std::string SystemConstants::General::COORDINATE_BASE = "Base"
const std::string SystemConstants::General::COORDINATE_WORLD = "Word"
const std::string SystemConstants::General::COORDINATE_TOOL = "Tool"
const std::string SystemConstants::General::COORDINATE_WOBJ = "Wobj"
```

Finally, I think that brackets should be added to the inner if-statements, even if they are single-lines. This so to be consistent with the rest of the code base.

Reply:
I made these changes in my latest commits. I also added an enumeration value `ACTIVE` to look up the `robtarget` relative to the active tool and coordinate frame. `ACTIVE` is the default coordinate for both the client and interface methods, so the methods should retain their previous behavior if no optional arguments are provided

Reply:
I'm also assuming that the string `Word` for world coordinates is not a typo as it appears in the RWS documentation

Reply:
> I'm also assuming that the string `Word` for world coordinates is not a typo as it appears in the RWS documentation

`Word` is not a typo, and I guessed that it corresponds to `World` since that is the only thing that makes sense to me. I don't know why the RWS developers implemented it like this.